app/src/main/java/com/ataulm/whatsnext/WhatsNextService.kt

```diff
@@ -17,24 +15,25 @@ internal class WhatsNextService(
 ) {
 
     suspend fun login(username: String, password: String): Token {
-        val authTokenApiResponse = letterboxdApi.fetchAuthToken(username, password)
-        return Token(authTokenApiResponse.accessToken, authTokenApiResponse.refreshToken)
+        return withContext(Dispatchers.IO) {
```

Comment:
We can inject the dispatchers at the service layer because we know that these network calls should be performed on the IO thread.

It means that our viewmodel scopes can use the Main dispatcher by default, and we don't have to explicitly specify `withContext` 

1) when we're doing heavy work (because it's specified here) and 
2) when we're switching back to do UI work (because it's on the correct dispatcher anyway)